### PR TITLE
add support for CUDA 12.1 and CUDA 12.2

### PR DESCRIPTION
**Pull request by pmeier**

### CODE REVIEW

File: light_the_torch/_cb.py [hunk 1]
```diff
@@ -176,6 +178,8 @@ def _detect_nvidia_driver_version() -> Optional[Version]:
     "Windows": {
         # Table 2 from
         # https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html
+        Version("12.2"): Version("525.41"),
+        Version("12.1"): Version("527.41"),
```
Comment (pmeier) [inline light_the_torch/_cb.py:182]: The lower driver version for a higher toolkit version is not a typo here, but maybe on the Nvidia side. I've asked for clarification.

Comment (jamesobutler) [reply light_the_torch/_cb.py:182]: @pmeier Indeed the lower driver version for 12.2 was observed on September 1st 2023 as seen at https://web.archive.org/web/20230901223827/https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html.

Reviewing the snapshot of the webpage on October 16th 2023, Nvidia had changed the version to 527.41 for CUDA 12.2 seemingly confirming that it was a typo on their side. See https://web.archive.org/web/20231016074125/https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html.

File: light_the_torch/_cb.py [hunk 2]
```diff
@@ -176,6 +178,8 @@ def _detect_nvidia_driver_version() -> Optional[Version]:
...
         Version("12.0"): Version("527.41"),
```
Comment (jamesobutler) [inline light_the_torch/_cb.py:183]: @pmeier Per the latest version of the webpage https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html
```suggestion
        Version("12.6"): Version("528.33"),
        Version("12.5"): Version("528.33"),
        Version("12.4"): Version("528.33"),
        Version("12.3"): Version("528.33"),
        Version("12.2"): Version("528.33"),
        Version("12.1"): Version("528.33"),
        Version("12.0"): Version("528.33"),
```

**Comment by jamesobutler** [comment]
@pmeier This PR has been superseded by CUDA 12.1-12.6 work included in #151.

**Pull request closed without merging**